zombies.js

```diff
@@ -78,8 +129,102 @@
  * @property {method} getPack              Returns private variable `pack`.
  * @property {method} getMaxHealth         Returns private variable `maxHealth`.
  */
-
-
+class Player {
+  constructor(name, health, strength, speed) {
+    this.name = name;
+    this.health = health;
+    this.strength = strength;
+    this.speed = speed;
+    this.isAlive = true;
+    this.equipped = false;
+    this._pack = []; // private
+    this._maxHealth = health; // private
+  }
+
+  getPack() {
+    // regular method
+    return this._pack;
+  }
+
+  getMaxHealth() {
+    return this._maxHealth;
+  }
+
+  takeItem(item) {
+    if (this.getPack().length < 3) {
+      // check if pack isn't full
+      console.log(Player.name + " picked up " + item.name);
+      return this.getPack().push(item); // add item to the pack
+    } else {
+      return console.log("Pack is full!");
+    }
+  }
+
+  discardItem(item) {
+    if (this.getPack().length === 0 || this.getPack().indexOf(item) === -1) {
```

Comment:
You can remove `this.getPack().length === 0` because checking the index of an item that doesn't exist in an array will return `-1` whether the array is empty or contains values.